### PR TITLE
Add implicit returns to styleguide

### PR DESCRIPTION
**Pull request by danspndl**
### Why?

Swift 5.1 allows implicit returns

 ### How?

- Added example for implicit returns for functions and computed properties

 ### PR checklist
* [x] I've assigned this PR to myself
* [x] I've set all the necessary PR's labels

 With :heart:

### CODE REVIEW

**Comment by Jimmybee** [comment]
So this implies using `return` on single expressions is wrong when it is *clear* (which is left up to who to decide?). I am concerned that this will result in lots of nits, when either is perfectly acceptable and should be left up the code author IMHO.

**Comment by sdaniel55** [comment]
> @Jimmybee: So this implies using `return` on single expressions is wrong when it is _clear_ (which is left up to who to decide?). I am concerned that this will result in lots of nits, when either is perfectly acceptable and should be left up the code author IMHO.

Both are acceptable, the goal would be to allow implicit returns only for single line expressions. I've made the changes more clear

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+### Why?
```
Comment (diegopetrucci) [inline .github/PULL_REQUEST_TEMPLATE.md]: Not sure if the template applies for playbook

**Review by diegopetrucci — DISMISSED** [review]
let's wait for banter

File: Cookbook/Style-guide/README.md [hunk 2]
```diff
@@ -381,10 +381,16 @@ Use self only when required by the compiler (in `@escaping` closures, or in init
 
 ## Computed Properties
 
-For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided.
+For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided. For single-expressions where the context is clear, both explicit and implicit returns can be used:
 
 **Preferred**:
 ```swift
+var diameter: Double {
```
Comment (ilyapuchka) [inline Cookbook/Style-guide/README.md]: how about `var diameter: Double { radius * 2 }`? should we allow this or discourage?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (inamiy) [reply .github/PULL_REQUEST_TEMPLATE.md]: PR template is unrelated to this PR's title, so pls propose in different PR.

File: Cookbook/Style-guide/README.md [hunk 3]
```diff
@@ -460,7 +468,21 @@ func updateConstraints() -> () {
 typealias CompletionHandler = (result) -> ()
 ```
 
-## Function Calls
+For single-expression functions where the context is clear, both explicit and implicit returns can be used:
```
Comment (ilyapuchka) [inline Cookbook/Style-guide/README.md]: This is similar to closures, do we have a preference to prefer no-return over return in closures? maybe then we should have the same preference here?

**Review by inamiy — CHANGES_REQUESTED** [review]
- [x] Resolves https://github.com/babylonhealth/ios-playbook/pull/267#discussion_r341116843

**Comment by ilyapuchka** [comment]
@Jimmybee we already have this situation with single-expression closures, but IIRC we prefer no-return over return in closures where it is possible. I think with should be consistent here with properties and method. Actually compiler will not compile if there is no return in the closure with more than one expression, at least it was like, not sure if anything changed with this swift proposal implemented and if it's the same with properties and methods

File: Cookbook/Style-guide/README.md (hunk 3 above)
Comment (mluisbrown) [reply Cookbook/Style-guide/README.md]: AFAIK we have a preference for not using `return` in single expression closures.

**Review by mluisbrown — APPROVED** [review]
I think it's enough that the styleguide *allows* not using `return`. However IMO it should not be a rule. So the current changes to the guide are good 👍

**Comment by mluisbrown** [comment]
> let's wait for banter

@diegopetrucci why? This is not a valid reason to block the PR IMO 👎

**Review by Jimmybee — APPROVED** [review]
For me, the style guide should be about improving readability of code. Allowing both doesn't detract from that.

File: Cookbook/Style-guide/README.md (hunk 2 above)
Comment (Jimmybee) [reply Cookbook/Style-guide/README.md]: It looks readable and easily understandable to me. I'd vote 👍

**Comment by mluisbrown** [comment]
@inamiy what changes are you requesting?

**Review by inamiy — APPROVED** [review]

File: Cookbook/Style-guide/README.md (hunk 2 above)
Comment (MichalTKwiecien) [reply Cookbook/Style-guide/README.md]: +1 for adding one-liner

Comment (sdaniel55) [reply Cookbook/Style-guide/README.md]: Added in f91870b

**Review by MichalTKwiecien — APPROVED** [review]

**Review by diegopetrucci — APPROVED** [review]

**Comment by sdaniel55** [comment]
Any objections? Or can I merge it? /cc @karabatov @ilyapuchka

**Review by ilyapuchka — APPROVED** [review]

**Review by karabatov — APPROVED** [review]
👍

**Pull request merged**